Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/mapping/CWAConfigMapper.kt

```diff
@@ -103,6 +84,31 @@ class CWAConfigMapper @Inject constructor() : CWAConfig.Mapper {
         }
     }
 
+    private fun ApplicationConfigurationAndroid.dccAdmissionCheckScenariosDisabled() = findBoolean(
+        labelValue = "dcc-admission-check-scenarios-disabled",
+        defaultValue = DCC_ADMISSION_CHECK_SCENARIOS_DISABLED
+    )
+
+    private fun ApplicationConfigurationAndroid.findBoolean(labelValue: String, defaultValue: Boolean): Boolean {
+        if (!hasAppFeatures()) return defaultValue
+
+        return try {
+            val value = (0 until appFeatures.appFeaturesCount)
+                .map { appFeatures.getAppFeatures(it) }
+                .firstOrNull { it.label == labelValue }
+                ?.value
+
+            return when (value) {
+                null -> defaultValue // no item found
+                1 -> true // found item has value '1'
+                else -> defaultValue // found item has value different to '1'
```

Comment:
The code that you posted has a problem that it returns `false` for every `it.value` different to `1` but it should return `defaultValue` in this case.

This was a very small bug that we had, but it wasn't a problem, since every default value that we have is `false`.